File: app/src/main/java/com/example/dreamloaf/data/ProductionDao.java
package com.example.dreamloaf.data;

import androidx.lifecycle.LiveData;
import androidx.room.Dao;
import androidx.room.Insert;
import androidx.room.Query;
import androidx.room.Update;

import java.util.List;

@Dao
public interface ProductionDao {
    @Insert
    void insert(Production production);

    @Update
    void update(Production production);

    @Query("DELETE FROM production WHERE id = :id")
    void delete(int id);

    @Query("SELECT * FROM production WHERE date = :date")
    LiveData<List<Production>> getProductionByDate(String date);

    @Query("SELECT * FROM production WHERE product_id = :productId AND date = :date LIMIT 1")
    Production getProductionForProductAndDate(int productId, String date);

    @Query("DELETE FROM production WHERE product_id = :productId")
    void deleteByProductId(int productId);

    @Query("SELECT * FROM production ORDER BY date DESC")
    LiveData<List<Production>> getAllProduction();
}

File: app/src/main/java/com/example/dreamloaf/data/ProductionDao.kt
package com.example.dreamloaf.data

import androidx.lifecycle.LiveData
import androidx.room.Dao
import androidx.room.Insert
import androidx.room.Query
import androidx.room.Update

@Dao
interface ProductionDao {
    @Insert
    fun insert(production: Production)

    @Update
    fun update(production: Production)

    @Query("DELETE FROM production WHERE id = :id")
    fun delete(id: Int)

    @Query("SELECT * FROM production WHERE date = :date")
    fun getProductionByDate(date: String?): LiveData<MutableList<Production?>?>?

    @Query("SELECT * FROM production WHERE product_id = :productId AND date = :date LIMIT 1")
    fun getProductionForProductAndDate(productId: Int, date: String?): Production?

    @Query("DELETE FROM production WHERE product_id = :productId")
    fun deleteByProductId(productId: Int)

    @get:Query("SELECT * FROM production ORDER BY date DESC")
    val allProduction: LiveData<MutableList<Production?>?>?
}
